sql/core/src/test/scala/org/apache/spark/sql/execution/OptimizeMetadataOnlyQuerySuite.scala

```diff
@@ -103,6 +103,21 @@ class OptimizeMetadataOnlyQuerySuite extends QueryTest with SharedSparkSession {
     "select partcol2, min(partcol1) from srcpart where partcol1 = 0 group by partcol2",
     "select max(c1) from (select partcol1 + 1 as c1 from srcpart where partcol1 = 0) t")
 
+  testMetadataOnly(
+    "SPARK-31590 " +
+      "The filter used by Metadata-only queries should filter out all the unevaluable expr",
```

Comment:
This isn't correct. Can you fix the title and scope it down only about subqueries?